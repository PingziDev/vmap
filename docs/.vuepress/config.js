const pkg = require('../../package.json')

module.exports = {
  title: 'Vmap',
  description: pkg.description,
  base: '/vmap/',
  themeConfig: {
    nav: [
      { text: '主页', link: '/' },
    ],
    sidebar: {
      '/introduction/': [
        {
          title: '快速上手',
          collapsable: false,
          children: [
            { title: '安装', path: 'install' },
            { title: '地图示例', path: 'map-demo' },
            { title: '标记示例', path: 'marker-demo' },
            { title: '地址搜索示例', path: 'search-demo' },
            { title: '定位示例', path: 'location-demo' },
            { title: '综合示例', path: 'composite-demo' },
          ],
        },
      ],
    },
    sidebarDepth: 1,
  },
  plugins: [
    // 回到顶部
    '@vuepress/back-to-top',
  ],
}
