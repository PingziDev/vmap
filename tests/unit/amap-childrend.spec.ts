import { createLocalVue, mount } from '@vue/test-utils'
import MaoLoader from '../../packages'
import { AMapConfig } from '../../examples/amap.config'

const localVue = createLocalVue()
localVue.use(MaoLoader, AMapConfig)

describe('map components', () => {
  const App = localVue.component('app', {
    data () {
      return {}
    },

    template: `
		<vue-amap>
			<vue-amap-marker></vue-amap-marker>
		</vue-amap>
    `,
  })

  const wrapper = mount(App)

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should inject parent', async () => {
    await localVue.nextTick()
    const provided = (wrapper.vm.$children[0] as any)._provided
    expect(provided.$amap).toBeDefined()
  })
})

