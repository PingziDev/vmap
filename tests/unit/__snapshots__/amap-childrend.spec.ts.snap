// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`map components should inject map 1`] = `
<div style="width: 500px; height: 500px;">
  <div style="width: 500px; height: 500px;"></div>
</div>
`;

exports[`map components should render correctly 1`] = `
<div style="width: 500px; height: 500px;">
  <div style="width: 500px; height: 500px;"></div>
</div>
`;
